src/app/services/configuration-updater/configurationUpdater.service.ts

```diff
@@ -0,0 +1,341 @@
+/*
+ * Licensed to Gisaïa under one or more contributor
+ * license agreements. See the NOTICE.txt file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Gisaïa licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { Aggregation } from 'arlas-api';
+
+@Injectable()
+export class ArlasConfigurationUpdaterService {
+
+  constructor() {
+  }
+
+  /**
+   * Parses the configuration and returns a list of contributors that should be removed from the config.
+   * @param data configuration object
+   * @param availableFields list of fields available for exploration
+   * @returns configuration object
+   */
+  public getContributorsToRemove(data, availableFields: Set<string>): Set<string> {
+    const contributorsToRemove = new Set<string>();
+    if (data) {
+      /** the conf is validated before; therefore, `arlas.web.contributors` is defined */
+      data.arlas.web.contributors.forEach(contributor => {
+        /** check if aggregation model has a non-available field (in bucket aggregation AND metrics) */
+        if (contributor.aggregationmodels) {
+          contributor.aggregationmodels.forEach((am: Aggregation) => {
+            if (!availableFields.has(am.field)) {
+              contributorsToRemove.add(contributor.identifier);
+            }
+            if (am.metrics) {
+              am.metrics.forEach(m => {
+                if (!availableFields.has(m.collect_field)) {
```

Comment:
Maybe you can handle it with a `fiter` instead of a `if` (also in other cases below)